x/evm/abci.go

```diff
@@ -506,6 +510,23 @@ func handleConfirmedEvents(ctx sdk.Context, bk types.BaseKeeper, n types.Nexus,
 	return nil
 }
 
+func setEventFailed(ctx sdk.Context, ck types.ChainKeeper, event types.Event, chainName nexus.ChainName) {
+	funcs.MustNoErr(ck.SetEventFailed(ctx, event.GetID()))
+
+	ck.Logger(ctx).Debug("failed handling event",
+		"chain", chainName,
+		"eventID", event.GetID(),
+	)
+
+	funcs.MustNoErr(ctx.EventManager().EmitTypedEvent(
+		&types.EVMEventFailed{
+			EventID: event.GetID(),
+			Chain:   chainName,
+		}),
+	)
```

Comment:
can just do these in the keeper method. Also, there's the `.Chain` in the event already so you don't need to pass in `chainName`

Reply:
updated